x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/serializer/serializer.ts

```diff
@@ -0,0 +1,161 @@
+  }
+
+  return produce<SerializedPolicy>(originalPolicy ?? defaultPolicy, (draft) => {
+    // Copy over all updated fields
+    merge(draft, updatedPolicy);
+
+    // Next copy over all meta fields and delete any fields that have been removed
+    // by fields exposed in the form. It is very important that we do not delete
+    // data that the form does not control! E.g., unfollow action in hot phase.
+
+    /**
+     * HOT PHASE SERIALIZATION
+     */
+    if (draft.phases.hot) {
+      draft.phases.hot.min_age = draft.phases.hot.min_age ?? '0ms';
+    }
+
+    if (draft.phases.hot?.actions) {
+      const hotPhaseActions = draft.phases.hot.actions;
+      if (hotPhaseActions.rollover && _meta.hot.useRollover) {
+        if (hotPhaseActions.rollover.max_age) {
+          hotPhaseActions.rollover.max_age = `${hotPhaseActions.rollover.max_age}${_meta.hot.maxAgeUnit}`;
+        }
+
+        if (hotPhaseActions.rollover.max_size) {
+          hotPhaseActions.rollover.max_size = `${hotPhaseActions.rollover.max_size}${_meta.hot.maxStorageSizeUnit}`;
+        }
+
+        if (!updatedPolicy.phases.hot!.actions?.forcemerge) {
+          delete hotPhaseActions.forcemerge;
+        } else if (_meta.hot.bestCompression) {
+          hotPhaseActions.forcemerge!.index_codec = 'best_compression';
+        }
+
+        if (_meta.hot.bestCompression && hotPhaseActions.forcemerge) {
+          hotPhaseActions.forcemerge.index_codec = 'best_compression';
+        }
+      } else {
+        delete hotPhaseActions.rollover;
+        delete hotPhaseActions.forcemerge;
+      }
+
+      if (!updatedPolicy.phases.hot!.actions?.set_priority) {
+        delete hotPhaseActions.set_priority;
+      }
+    }
+
+    /**
+     * WARM PHASE SERIALIZATION
+     */
+    if (_meta.warm.enabled) {
+      const warmPhase = draft.phases.warm!;
+      // If warm phase on rollover is enabled, delete min age field
+      // An index lifecycle switches to warm phase when rollover occurs, so you cannot specify a warm phase time
+      // They are mutually exclusive
+      if (
+        (!_meta.hot.useRollover || !_meta.warm.warmPhaseOnRollover) &&
+        updatedPolicy.phases.warm!.min_age
+      ) {
+        warmPhase.min_age = `${updatedPolicy.phases.warm!.min_age}${_meta.warm.minAgeUnit}`;
+      } else {
+        delete warmPhase.min_age;
+      }
+
+      warmPhase.actions = serializeMigrateAndAllocateActions(
+        _meta.warm,
+        warmPhase.actions,
+        originalPolicy?.phases.warm?.actions
+      );
+
+      if (!updatedPolicy.phases.warm!.actions?.forcemerge) {
+        delete warmPhase.actions.forcemerge;
+      } else if (_meta.warm.bestCompression) {
+        warmPhase.actions.forcemerge!.index_codec = 'best_compression';
+      }
+
+      if (!updatedPolicy.phases.warm!.actions?.set_priority) {
+        delete warmPhase.actions.set_priority;
+      }
+
+      if (!updatedPolicy.phases.warm!.actions?.shrink) {
+        delete warmPhase.actions.shrink;
```

Comment:
This line too.

Also, I don't see this corresponding logic in the original `createSerializer`. Is this newly added in this PR?

Reply:
It's needed after we changed the strategy of merging with previously set values to ensure that we remove it. Also this work has not been released so we don't need to add fix release note :)